unison-cli/src/Unison/Codebase/Editor/HandleInput/ProjectSwitch.hs

```diff
@@ -33,25 +34,33 @@ projectSwitch projectAndBranchNames0 = do
   let projectId = project ^. #projectId
   maybeCurrentProject <- ProjectUtils.getCurrentProjectBranch
   (outcome, branchId) <-
-    Cli.runEitherTransaction do
+    Cli.runTransaction do
       Queries.loadProjectBranchByName projectId branchName >>= \case
-        Just branch -> pure (Right (SwitchedToExistingBranch, branch ^. #branchId))
+        Just branch -> pure (SwitchedToExistingBranch, branch ^. #branchId)
         Nothing -> do
-          -- We don't support creating a new branch from outside its project, because we want to require that the user
-          -- picks some parent branch to start from
-          case maybeCurrentProject of
-            Just (ProjectAndBranch currentProject currentBranch) | projectId == currentProject ^. #projectId -> do
-              newBranchId <- Sqlite.unsafeIO (ProjectBranchId <$> UUID.nextRandom)
-              Queries.insertProjectBranch projectId newBranchId branchName
-              Queries.markProjectBranchChild projectId (currentBranch ^. #branchId) newBranchId
-              pure (Right (SwitchedToNewBranchFrom currentBranch, newBranchId))
-            _ -> pure (Left (Output.RefusedToCreateProjectBranch projectAndBranchNames))
+          newBranchId <- Sqlite.unsafeIO (ProjectBranchId <$> UUID.nextRandom)
+          -- If we create a new branch from outside its project, then the new branch is "detached" (no history, no
+          -- parent).
+          let parentBranch = do
+                ProjectAndBranch currentProject currentBranch <- maybeCurrentProject
+                guard (projectId == currentProject ^. #projectId)
```

Comment:
Just a thought, but maybe we should allow creating branches in different projects if it's done explicitly; In Git I'd just copy-paste on the file-system but that's harder to do in UCM.

That said, I dunno if there's really any practical use-case for it so I don't particularly care either way 🤷🏼‍♂️